frontend/packages/dev-console/src/components/topology2/componentUtils.ts

```diff
@@ -73,14 +74,16 @@ const nodeDragSourceSpec = (
   allowRegroup: boolean = true,
 ): DragSourceSpec<DragObjectWithType, Node, {}, NodeProps> => ({
   item: { type },
-  operation: allowRegroup
-    ? {
-        [Modifiers.SHIFT]: REGROUP_OPERATION,
-      }
-    : undefined,
+  operation: (monitor, props) => {
+    return props.canRegroup && allowRegroup
+      ? {
+          [Modifiers.SHIFT]: REGROUP_OPERATION,
+        }
+      : undefined;
+  },
   canCancel: (monitor) => monitor.getOperation() === REGROUP_OPERATION,
   end: async (dropResult, monitor, props) => {
-    if (!monitor.isCancelled() && monitor.getOperation() === REGROUP_OPERATION) {
+    if (monitor.getOperation() === REGROUP_OPERATION) {
```

Comment:
Was it intentional to remove the check for cancelled?

Reply:
No @jeff-phillips-18. My bad. Reverted it back.